### PR TITLE
 ✨ Add configurable workers for individual controllers, defaults to 10

### PR DESCRIPTION
**Pull request by dilipmighty245**
## Summary
To make the worker count configurable for individual controllers which can be leveraged in scale environment

Related issue(s)
Fixes #

## Release Notes
Set worker count configurable, defaults to 10

### CODE REVIEW

**Comment by mjudeikis** [comment]
Can we do this more generic:
`https://kubernetes.slack.com/archives/C021U8WSAFK/p1710515852080189?thread_ts=1710423745.848959&cid=C021U8WSAFK` ?

File: cmd/kcp/help_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2022 The KCP Authors.
```
Comment (sttts) [inline cmd/kcp/help_test.go:2]: 2024

File: cmd/kcp/kcp.go [hunk 2]
```diff
@@ -76,6 +77,10 @@ func main() {
 		} else if strings.HasPrefix(f, "--root-directory=") {
 			rootDir = strings.TrimPrefix(f, "--root-directory=")
 		}
+
+		if strings.HasPrefix(f, "--controller-threads=") {
```
Comment (sttts) [inline cmd/kcp/kcp.go:81]: workers is the normal wording, not threads.

File: pkg/server/options/options.go [hunk 3]
```diff
@@ -197,6 +197,11 @@ Prefixing with - or + means to remove from the default set or add to the default
 	fs.Var(kcpfeatures.NewFlagValue(), "feature-gates", ""+
 		"A set of key=value pairs that describe feature gates for alpha/experimental features. "+
 		"Options are:\n"+strings.Join(kcpfeatures.KnownFeatures(), "\n")) // hide kube-only gates
+
+	// add flags that are filtered out from upstream, but overridden here with our own version
+	fs.Var(kcpfeatures.NewFlagValue(), "controller-settings", ""+
+		"A set of key=value pairs that manage controller settings for alpha/experimental features. "+
+		"Options are:\n"+strings.Join(kcpfeatures.KnownFeatures(), "\n")) // hide kube-only gates
```
Comment (sttts) [inline pkg/server/options/options.go:204]: not sure I understand from the description and go doc what this is.

**Pull request closed without merging**